disperser/apiserver/disperse_blob_v2.go

```diff
@@ -70,33 +70,40 @@ func (s *DispersalServerV2) StoreBlob(ctx context.Context, data []byte, blobHead
 func (s *DispersalServerV2) validateDispersalRequest(req *pb.DisperseBlobRequest) error {
 	data := req.GetData()
 	blobSize := len(data)
-	if uint64(blobSize) > s.maxNumSymbolsPerBlob*encoding.BYTES_PER_SYMBOL {
-		return api.NewErrorInvalidArg(fmt.Sprintf("blob size cannot exceed %v bytes", s.maxNumSymbolsPerBlob*encoding.BYTES_PER_SYMBOL))
-	}
 	if blobSize == 0 {
 		return api.NewErrorInvalidArg("blob size must be greater than 0")
 	}
+	blobLength := encoding.GetBlobLengthPowerOf2(uint(blobSize))
+	if blobLength > uint(s.maxNumSymbolsPerBlob) {
+		return api.NewErrorInvalidArg("blob size too big")
+	}
 
 	blobHeaderProto := req.GetBlobHeader()
 	if blobHeaderProto.GetCommitment() == nil {
 		return api.NewErrorInvalidArg("blob header must contain commitments")
 	}
 
+	if len(blobHeaderProto.GetQuorumNumbers()) == 0 {
+		return api.NewErrorInvalidArg("blob header must contain at least one quorum number")
+	}
+
 	if len(blobHeaderProto.GetQuorumNumbers()) > int(s.onchainState.QuorumCount) {
 		return api.NewErrorInvalidArg(fmt.Sprintf("too many quorum numbers specified: maximum is %d", s.onchainState.QuorumCount))
 	}
 
+	for _, quorum := range blobHeaderProto.GetQuorumNumbers() {
+		if quorum > corev2.MaxQuorumID || uint8(quorum) > s.onchainState.QuorumCount {
+			return api.NewErrorInvalidArg(fmt.Sprintf("invalid quorum number %d; maximum is %d", quorum, s.onchainState.QuorumCount))
+		}
```

Comment:
The quorum ID in the protobuf is `uint32`, why do you cast it into a `uint8`?

Do quorum numbers start at `0` or `1`? If they start at `0`, should it instead be `>= s.onchainState.QuorumCount`?

Reply:
> The quorum ID in the protobuf is uint32, why do you cast it into a uint8?

It needs to be converted to compare to `s.onchainState.QuorumCount`, which is `uint8`

> Do quorum numbers start at 0 or 1? If they start at 0, should it instead be >= s.onchainState.QuorumCount?

Nice catch! Will update.